sui/src/unit_tests/rpc_server_tests.rs

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::net::SocketAddr;
+use std::path::Path;
+use std::str::FromStr;
+
+use jsonrpsee::http_client::HttpClientBuilder;
+use jsonrpsee::http_server::{HttpServerBuilder, HttpServerHandle};
+use move_core_types::identifier::Identifier;
+
+use sui::config::{PersistedConfig, WalletConfig};
+use sui::keystore::{Keystore, SuiKeystore};
+use sui::rest_gateway::responses::ObjectResponse;
+use sui::rpc_gateway::RpcGatewayServer;
+use sui::rpc_gateway::TransactionBytes;
+use sui::rpc_gateway::{Base64EncodedBytes, RpcGatewayClient};
+use sui::rpc_gateway::{RpcGatewayImpl, SignedTransaction};
+use sui::sui_json::{resolve_move_function_args, SuiJsonValue};
+use sui::{SUI_GATEWAY_CONFIG, SUI_WALLET_CONFIG};
+use sui_core::gateway_state::gateway_responses::TransactionResponse;
+use sui_framework::build_move_package_to_bytes;
+use sui_types::base_types::ObjectID;
+use sui_types::object::ObjectRead;
+use sui_types::SUI_FRAMEWORK_ADDRESS;
+
+use crate::rpc_server_tests::sui_network::start_test_network;
+
+mod sui_network;
+
+#[tokio::test]
+async fn test_get_objects() -> Result<(), anyhow::Error> {
+    let temp_dir = tempfile::tempdir()?;
+    let working_dir = temp_dir.path();
+    let _network = start_test_network(working_dir, None).await?;
+    let (server_addr, _handle) = start_rpc_gateway(&working_dir.join(SUI_GATEWAY_CONFIG)).await?;
+    let wallet_conf: WalletConfig = PersistedConfig::read(&working_dir.join(SUI_WALLET_CONFIG))?;
+    let address = wallet_conf.accounts.first().unwrap();
```

Comment:
This snippet is repeated multiple times within this file, can we put them into a helper function like the factoring done in https://github.com/MystenLabs/sui/pull/1412/files#diff-c327b473970c4626b8103cda98ad7b8f17fdd40d7d603f42821f2f609fd1b314 ?